onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -567,26 +581,38 @@ ORT_API_STATUS_IMPL(OrtApis::GetStringTensorDataLength, _In_ const OrtValue* val
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::GetStringTensorElementLength, _In_ const OrtValue* value, _Out_ size_t* out, size_t index) {
+  TENSOR_READ_API_BEGIN
+  const auto* src = tensor.Data<std::string>();
+  auto len = tensor.Shape().Size();
+  if (index < len) {
+    *out = src[index].size();
+  } else
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "shape is invalid");
+  return nullptr;
+  API_IMPL_END
+}
+
 ORT_API_STATUS_IMPL(OrtApis::GetStringTensorContent, _In_ const OrtValue* value, _Out_writes_bytes_all_(s_len) void* s,
                     size_t s_len, _Out_writes_all_(offsets_len) size_t* offsets, size_t offsets_len) {
   TENSOR_READ_API_BEGIN
   const auto* input = tensor.Data<std::string>();
   auto len = static_cast<size_t>(tensor.Shape().Size());
   if (offsets_len < len) {
-    return OrtApis::CreateStatus(ORT_FAIL, "space is not enough");
+    return OrtApis::CreateStatus(ORT_FAIL, "offsets buffer is too small");
   }
   {
     size_t ret = 0;
     for (size_t i = 0; i != len; ++i) {
       ret += input[i].size();
     }
     if (s_len < ret) {
-      return OrtApis::CreateStatus(ORT_FAIL, "space is not enough");
+      return OrtApis::CreateStatus(ORT_FAIL, "output buffer is too small");
     }
   }
   size_t f = 0;
   char* p = static_cast<char*>(s);
-  for (size_t i = 0; i != offsets_len; ++i, ++offsets) {
+  for (size_t i = 0; i != len; ++i, ++offsets) {
```

Comment:
Interesting, if offsets_len is larger than necessary this would have read beyond the valid members. But now it's leaving the remaining offsets uninitialized? I wonder if it's better to require offsets_len to match exactly.

Reply:
Agreed, can add an error check here for `!=` instead of : `offsets_len < len`.